src/commands/ticketing/embed-ticket.ts

```diff
@@ -0,0 +1,179 @@
+		)
+		.addChannelOption((option) =>
+			option
+				.setName('channel')
+				.setDescription('The channel where the ticket embed will be in')
+				.setRequired(true)
+				.addChannelTypes([ChannelType.GuildText, ChannelType.GuildNews])
+		),
+	execute: async function ({ interaction }) {
+		try {
+			if (
+				!interaction.memberPermissions?.has(['MANAGE_CHANNELS', 'MANAGE_GUILD'])
+			) {
+				return interaction.reply({
+					content:
+						'You need the manage channels and guild permission to run this command',
+					ephemeral: true
+				});
+			}
+
+			const [rows] = await conn.execute(
+				'SELECT * FROM TicketingManagers WHERE GuildID = ?',
+				[interaction.guildId]
+			);
+			const record = (
+				rows as RowDataPacket[]
+			)[0] as Tables.TicketingManagers | null;
+
+			if (!record) {
+				return interaction.reply({
+					content: 'No ticket record could be found, please create one',
+					ephemeral: true
+				});
+			}
+
+			const channel = interaction.options.getChannel('channel')! as
+				| TextChannel
+				| NewsChannel;
+
+			if (
+				!channel.permissionsFor(interaction.guild!.me!).has(['SEND_MESSAGES'])
+			) {
+				return interaction.reply({
+					content: 'I need the send messages permission for that channel',
+					ephemeral: true
+				});
+			}
+
+			const notes = [
+				'You can only have one ticket opened at a time',
+				'By using the button, the bot will not be able to provide the subject of the ticket',
+				'You can use the buttons in the ticket channel to delete (and archive if in a thread) the ticket'
+			].map((txt) => `- ${txt}`);
+
+			const embed = new MessageEmbed()
+				.setColor('RANDOM')
+				.setTitle('Create Support Ticket')
+				.addField(
+					'Usage',
+					`Create a support ticket by either clicking the button below, or by using the command ${inlineCode(
+						'/ticket'
+					)}`
+				)
+				.addField('Notes', notes.join('\n'))
+				.setTimestamp();
+			const row = new MessageActionRow().addComponents(
+				new MessageButton()
+					.setCustomId(this.components!.customIds![0])
+					.setStyle('PRIMARY')
+					.setEmoji('🎟️')
+					.setLabel('Create Ticket')
+			);
+
+			// text channel based ticketing
+			if (record.SupportCategory !== '0' && record.UseTextChannels) {
+				const msg = await channel.send({ embeds: [embed], components: [row] });
+				await msg.pin();
+
+				if (channel.lastMessage?.system && channel.lastMessage.deletable) {
+					await channel.lastMessage.delete();
+				}
+
+				interaction.reply({
+					content: 'Ticket embed successfully sent!',
+					ephemeral: true
+				});
+			} else {
+				const supportChannelWithoutRecord =
+					interaction.guild!.channels.cache.find(
+						(channel) =>
+							(channel.name === 'support' && channel.type === 'GUILD_TEXT') ||
+							channel.type === 'GUILD_NEWS'
+					);
+				const supportChannelWithRecord =
+					await interaction.guild!.channels.fetch(record.SupportChannel);
```

Comment:
This could error since the if statement is checking for a support category and not a support channel (it can be '0')

```suggestion
				const supportChannelWithRecord =
					interaction.guild!.channels.cache.get(record.SupportChannel);
```